src/app/src/main/java/com/example/bookmark/MyBooksActivity.java

```diff
@@ -55,57 +45,54 @@ public void onClick(View v) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_my_books);
-        getSupportActionBar().setTitle("My Books");
-
-        booksListView = findViewById(R.id.my_books_listview);
 
-        addBookBtn = findViewById(R.id.my_books_add_btn);
+        // Add the add book button
+        addBookBtn = findViewById(R.id.listing_books_action_btn);
+        addBookBtn.setVisibility(View.VISIBLE);
         addBookBtn.setOnClickListener(addBookListener);
-        booksAdapter = new BookList(this, filteredBooks, false, true);
-        booksListView.setAdapter(booksAdapter);
-
-        // Fetch users books
-        getBooks();
-
-        booksListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
-
-            @Override
-            public void onItemClick(AdapterView<?> parent, View view,
-                                    int position, long id) {
-                Bundle bundle = new Bundle();
-                bundle.putSerializable("Book", filteredBooks.get(position));
-                bundle.putSerializable("User", user);
-                Intent intent = new Intent(MyBooksActivity.this, MyBookDetailsActivity.class);
-                intent.putExtras(bundle);
-
-                startActivity(intent);
-            }
-
-        });
-
-        // Pull down to refresh
-        final SwipeRefreshLayout pullToRefresh = findViewById(R.id.swiperefresh);
-        pullToRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
-            @Override
-            public void onRefresh() {
-                getBooks();
-                pullToRefresh.setRefreshing(false);
-            }
-        });
-
     }
 
     @Override
     protected void onResume() {
         super.onResume();
-        getBooks();
+        getRelevantBooks();
```

Comment:
I think ListingBooksActivity would already call getRelevantBooks() from within its getBooks() call that happens within the ListingBooksActivity.onResume() method